src/shipit_code_coverage/shipit_code_coverage/codecov.py

```diff
@@ -49,7 +50,11 @@ def download_coverage_artifacts(build_task_id):
 
 def get_github_commit(mercurial_commit):
     url = 'https://api.pub.build.mozilla.org/mapper/gecko-dev/rev/hg/%s'
-    r = requests.get(url % mercurial_commit)
+    while True:
```

Comment:
Maybe you need to set a maximum allowed time to wait (i'm a bit afraid of a while True: sleep :fearful: )